optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/spotroster/ShiftBlobPopoverContent.java

```diff
@@ -0,0 +1,217 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.openshift.employeerostering.gwtui.client.pages.spotroster;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import elemental2.dom.Event;
+import elemental2.dom.HTMLButtonElement;
+import elemental2.dom.HTMLDivElement;
+import elemental2.dom.HTMLElement;
+import elemental2.dom.HTMLInputElement;
+import elemental2.dom.MouseEvent;
+import org.gwtbootstrap3.client.ui.ListBox;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.ForEvent;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.powers.BlobPopover;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.powers.BlobPopoverContent;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.view.BlobView;
+import org.optaplanner.openshift.employeerostering.shared.employee.Employee;
+import org.optaplanner.openshift.employeerostering.shared.employee.EmployeeRestServiceBuilder;
+import org.optaplanner.openshift.employeerostering.shared.shift.Shift;
+import org.optaplanner.openshift.employeerostering.shared.shift.ShiftRestServiceBuilder;
+import org.optaplanner.openshift.employeerostering.shared.shift.view.ShiftView;
+import org.optaplanner.openshift.employeerostering.shared.spot.SpotRestServiceBuilder;
+
+import static java.lang.Long.parseLong;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+import static org.optaplanner.openshift.employeerostering.gwtui.client.common.FailureShownRestCallback.onSuccess;
+
+@Templated
+public class ShiftBlobPopoverContent implements BlobPopoverContent {
+
+    @Inject
+    @DataField("root")
+    private HTMLDivElement root;
+
+    @Inject
+    @DataField("close-button")
+    private HTMLButtonElement closeButton;
+
+    @Inject
+    @DataField("from-day")
+    private HTMLInputElement fromDay;
+
+    @Inject
+    @DataField("from-hour")
+    private HTMLInputElement fromHour;
+
+    @Inject
+    @DataField("to-day")
+    private HTMLInputElement toDay;
+
+    @Inject
+    @DataField("to-hour")
+    private HTMLInputElement toHour;
+
+    @Inject
+    @DataField("spot")
+    private ListBox spotSelect; //FIXME: Don't use GWT widget
```

Comment:
I suggest MultipleSelect from gwtbootscrapt. It what I used in the Employee/Spot screens (see SpotSubform)

Reply:
Oh right... Select from gwtbootscrapt (same thing, but only for single things XD)